sos/report/plugins/azure.py

```diff
@@ -66,4 +67,21 @@ def setup(self):
                 'openssl x509 -noout -text -in ' + crt_path
             ])
 
+
+class AzureEnhancedMonitoring(Azure, RedHatPlugin):
+
+    def setup(self):
+        super(AzureEnhancedMonitoring, self).setup()
+        aem_settings = '/var/lib/waagent/' \
+            'Microsoft.OSTCExtensions.AzureEnhancedMonitorForLinux-*/' \
+            'config/0.settings'
+        # Check if AEM (Azure Enhanced Monitoring for Linux) is active
+        aem_glob = glob.glob(aem_settings)
+        if aem_glob:
+            if os.path.isfile(aem_glob[0]):
+                # Add AEM PerfCounters
```

Comment:
What are we guarding against here? Presumably there are non-file type path entries (directories? sockets?) that cannot be discriminated by pattern and that mean that you don't want to use the automatic glob expansion in `add_copy_spec()` here - that kind of deviation from the normal flow should have a comment explaining why it's being done (big subdirectories, socket files or whatever we're trying to avoid collecting).

In other words what is the condition that could cause `if os.path.isfile(aem_glob[0])` to be `False`?

Reply:
AEM may not be available and then this file doesn't exist AFAIK.
So, are you suggesting to just do a `add_copy_spec('/var/lib/AzureEnhancedMonitor/')`?

Reply:
All collection in `sos` is speculative - if paths don't exist then that's fine (and explicitly not an error). This is how we support components that have optional features, or feature variation across versions.

It makes the code more complicated to review and maintain to put these extra checks in the plugin so I would suggest not doing that unless there is a good reason to (like for e.g. huge files or subdirectories that cannot be excluded by the glob pattern).

Reply:
I get your point.
So I'm better off by adding `var/lib/AzureEnhancedMonitor/` to `add_copy_spec` in the Azure class, right?

Reply:
So long as that is safe yes - the only reason to elevate glob handling out of `add_copy_spec()` and into the plugin itself is if there is special logic needed to skip certain paths due to size or other side-effects. If it's just performance counters and they are reasonably sized (and don't scale to 1000s of instances or whatever) then it's fine to just try to scoop them all up in one go.

Btw isn't the path `/var/lib/waagent/Microsoft.OSTCExtensions.AzureEnhancedMonitorForLinux-*`?

Reply:
Got it. Then it will boil down to a single line commit :-P
There are two paths:

    /var/lib/waagent/Microsoft.OSTCExtensions.AzureEnhancedMonitorForLinux-*/config/0.settings

for determining if AEM is enabled on that host

and 

    /var/lib/AzureEnhancedMonitor/

containing the data.

Reply:
Ah, gotcha - thanks - I didn't look closely at the path in the `add_copy_spec()`.

Reply:
Only one question left... The paths in `add_copy_spec` are currently using double quotes, although not necessary. If I add my path now, shall I add it with double quotes, with single quote and if the later, shall I change the other paths also?

Reply:
We've not been all that consistent on this down the years so the general rule of thumb (unless the contributor guide has been updated & I missed it) is to go with the style of the code you are changing - so double quotes in this case.

For new plugins I would tend to prefer double quotes - simply because there are more of them in the tree right now, so when it does eventually get cleaned up there should be fewer single quote cases to fix:

```
$ git grep "add_copy_spec('" | wc -l
35
$ git grep 'add_copy_spec("' | wc -l
152
```

Reply:
Ack. Then I'll change the code accordingly - add the single line with double quotes!
Thanks for your help @bmr-cymru !